### ISSUE TITLE
Script broken with new Cloudflare protections

### ISSUE THREAD

**Issue by Kyther** [nianeyna/ao3downloader#96]
No idea if this is fixable given the protections are trying to stop a DDoS attack, but I decided to test with a small fandom to see if it would be able to download anything.

I get:

`'NoneType' object has no attribute 'find'`

And I'm bumped back to the main menu of the script.

**Comment by WhyDidIHaveToDoThis** [comment]
This is a tragic day ;(
Cloudflare on ffnet and now on ao3.

**Comment by Lagicrus** [comment]
Try testing again? With another script that was broken due to CF, it seems to now be working fine as I suspect they lowered protections again? Maybe this will work again with no issues

**Comment by nianeyna** [comment]
Yes, I believe they have turned off the bot challenge, but it's still looking pretty broken to me. I am not sure if that has to do with cloudflare or if it's just general bugginess from the DDoS. I'm going to leave the notice in the readme up for now.

**Comment by Kyther** [comment]
I'm still having to click "I'm a human" periodically just browsing the site - no captchas, but have to click in the box - so it's definitely going to be borked until things get lowered some.  I guess wait and see…  Adapting the script to the protections would be possible but the rate limits are so extreme right now it would be an exercise in frustration, I think.

**Comment by nianeyna** [comment]
Right, I think there's not much point in trying to adapt the script at this stage - things are still changing by the minute, even if I were to fix the problem it might be broken again before I finished pushing the fix, haha. I'm keeping an eye out for developments but for now my plan is pretty much to sit tight and wait it out.

**Comment by Kyther** [comment]
The restrictions are definitely fluctuating.  I tested it on a small fandom this evening and it actually worked!  The break periods are wildly different, though - I got 17 seconds the first time and 303 the next, lol.  But the automation is the important thing, for me - the time it takes isn't so big a deal because I'm not having to manually click through it all.

I'm afraid to use it on the really big one I'd had in my queue next, in case they suddenly increase the restrictions to deal with stuff and it fails mid-script.  I'm *hoping* if that's the case, that I'd be able to choose 'y' in the "do you want to start downloading from the page you stopped on last time'.  The other thought that occurred to me is this is where I might do best grabbing all links to a file and then breaking that up, feeding only a chunk at a time to the script so if it fails, I'll know where.  (Or will it tell me which link it was trying to get when it failed?  I've not actually tried the links to file and download links from file options.)  Any ideas for the best way to proceed without risking requesting the same fics from AO3 twice? :)

**Comment by Kyther** [comment]
Thanks for the warning!  Other than the small fandom - which the number of files definitely matched with the general expected number - I've been grabbing all links to a file and splitting it up into txt files of 500 links apiece, then running those through the script one at a time.  It's stabilized at 300 secs break every time, which is generally every 27-30 downloads.  I checked the numbers I should have against what I actually do have and so far just one hasn't downloaded.  Probably means it's the final link on one of the completed fics so far, which would narrow things down considerably - or it means that one of the fics was deleted between when I grabbed the links and when I downloaded the fics from the list of links.  I'll check the final links to see, but if it's not them, will have to assume the fic I was aiming for is deleted.
